src/net.c

```diff
@@ -1795,15 +1776,27 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
   coap_option_filter_clear(opt_filter);
 
   /* try to find the resource from the request URI */
-  coap_hash_request_uri(pdu, key);
-  resource = coap_get_resource_from_key(context, key);
+  str *uri_path = coap_get_uri_path(pdu);
+  if (!uri_path)
+    return;
+  resource = coap_get_resource_from_uri_path(context, *uri_path);
 
-  if (!resource) {
+  if (!resource || resource->unknown) {
```

Comment:
Actually unknown is a single bit - I have updated it to be known as is_unknown for clarity.  The bit is there so that if anyone actually does a resource request against '- Unknown -', we do not treat it as a normal resource.
I have changed it to
if ((resource == NULL) || (resource->is_unknown == 1)) {